onboarding_auth/onboardingAuthDomain/src/main/java/com/alexafit/onboardingauthdomain/useCase/ValidateEmail.kt

```diff
@@ -0,0 +1,19 @@
+package com.alexafit.onboardingauthdomain.useCase
+
+class ValidateEmail {
+
+    operator fun invoke(email: String): Pair<String, Boolean> {
```

Comment:
Not a big fan of Pairs since they don't add any readability/context to your code. To understand what this returns, you would need to read all the code of this use case. Instead, I'd recommend to work with a result class:
```
data class ValidateEmailResult(
    val trimmedEmail: String,
    val isValid: String
)
```